jobs-service/jobs-service-pg/src/main/java/org/kie/kogito/jobs/service/repository/pg/marshaller/TriggerMarshaller.java

```
@@ -0,0 +1,189 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.kogito.jobs.service.repository.pg.marshaller;
+
+import java.util.Date;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.kie.kogito.timer.Trigger;
+import org.kie.kogito.timer.impl.IntervalTrigger;
+import org.kie.kogito.timer.impl.PointInTimeTrigger;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.json.JsonObject;
+
+@ApplicationScoped
+public class TriggerMarshaller {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TriggerMarshaller.class);
+
+    private static final String CLASS_TYPE = "classType";
+
+    public JsonObject marshall(Trigger trigger) {
+        return Stream.of(
+                Map.entry(
+                        (Predicate<Trigger>) IntervalTrigger.class::isInstance,
+                        (Function<Trigger, JsonObject>) t -> JsonObject.mapFrom(new IntervalTriggerAccessor((IntervalTrigger) t))
+                                .put(CLASS_TYPE, t.getClass().getName())),
+                Map.entry(
+                        (Predicate<Trigger>) PointInTimeTrigger.class::isInstance,
+                        (Function<Trigger, JsonObject>) t -> JsonObject.mapFrom(new PointInTimeTriggerAccessor((PointInTimeTrigger) t))
+                                .put(CLASS_TYPE, t.getClass().getName())))
+                .filter(e -> e.getKey().test(trigger))
+                .map(e -> e.getValue().apply(trigger))
+                .findAny()
+                .orElseGet(() -> {
+                    LOGGER.debug("Cannot marshall Trigger to Json: {}", trigger);
```

should we throw an exception if this happens? Same for unmarshall. @xieshenzh @tiagodolphine wdyt?